lib/arproxy/proxy_chain.rb

```diff
@@ -35,13 +35,28 @@ def execute_with_arproxy(sql, name=nil, **kwargs)
         end
         alias_method :execute_without_arproxy, :execute
         alias_method :execute, :execute_with_arproxy
+
+        if private_method_defined?(:raw_execute) # Rails 7.1+
+          private def raw_execute_with_arproxy(sql, name, **kwargs)
+            ::Arproxy.proxy_chain.connection = self
+            ::Arproxy.proxy_chain.head.send :raw_execute, sql, name, **kwargs
+          end
+          alias_method :raw_execute_without_arproxy, :raw_execute
+          alias_method :raw_execute, :raw_execute_with_arproxy
```

Comment:
Doesn't this break codes that directly call `execute`, which might call `raw_execute` internally, by invoking proxy_chain twice?

Perhaps the specs also need changes, since we only have DummyAdapter, which has independent `execute` and `raw_execute`.

Reply:
Are you talking about the case where both `execute` and `raw_execute` are defined?
I am assuming that only one of them is defined based on the version of ActiveRecord you are using.

Do you have ideas for a more approach in changing use?

Reply:
Both of the two methods should be defined in connection adapters. For example, Mysql2Adapter has both of them:

```ruby
% bin/rails c
Loading development environment (Rails 7.1.4)
irb(main):001> ActiveRecord::ConnectionAdapters::Mysql2Adapter.instance_methods.include?(:execute)
=> true
irb(main):002> ActiveRecord::ConnectionAdapters::Mysql2Adapter.private_instance_methods.include?(:raw_execute)
=> true
```

Also `ActiveRecord::ConnectionAdapters::DatabaseStatements` defines both two: https://github.com/rails/rails/blob/v7.1.4/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

Reply:
Thank you for more information.
I got to make sense of what you pointed out. :smile:
I will try to change the code.